Autolog with PyTorch using lightning - Examples (#3608)
mlflow/mlflow: examples/pytorch/MNIST/example2/README.md

> ## MNIST example with MLFlow

Reviewer: @shrinath-suresh what's the difference between this example (example2) and example1? Could we include just one or the other?

Reviewer: Also, is this example adapted from an existing PytorchLightning example somewhere? Maybe we could reference the existing example code, and mention ("This example is adapted from ... with only minor tweaks, i.e. the addition of an `mlflow.pytorch.autolog` call to add automatic instrumentation of training code")

Author: > @shrinath-suresh what's the difference between this example (example2) and example1? Could we include just one or the other?

Example 1 illustrates - EarlyStopping Behaviour, Model Checkpointing, Learning Rate Monitor
Example 2 illustrates - Logging every n iterations

Author: > Also, is this example adapted from an existing Pytorch Lightning example somewhere? Maybe we could reference the existing example code, and mention ("This example is adapted from ... with only minor tweaks, i.e. the addition of an `mlflow.pytorch.autolog` call to add automatic instrumentation of training code")

sure. i will add a link to it

> ## MNIST example with MLFlow

Reviewer: cc @andreakress would you be able to help take a look at the README here for wording?

> In this example, we train a model to predict handwritten digits.The autolog code uses Pytorch Lightning's MLFlowLogger to log metrics. 

Reviewer: ```suggestion
In this example, we train a Pytorch Lightning model to predict handwritten digits. The code is almost entirely dedicated to model training, with the addition of a single ``mlflow.pytorch.autolog()`` call to enable automatic logging of params, metrics, and models from training. 
```

I think we could reword the overview as in the above suggestion^

The code is trained using pytorch lightning loop and the autolog function call in the main - `autolog()`
is responsible for logging the params, metrics, model summary and the model.
This example logs metrics only after n epoch iterations. The iteration limit can be set in the autolog method using the parameter `log_every_n_iter=NUMBER-OF-ITERATIONS`.

> For ex: `autolog(log_every_n_iter=5)`

Reviewer: Could we remove this in favor of linking to the ``mlflow.pytorch.autolog`` doc once it's available, as described in https://github.com/mlflow/mlflow/pull/3608/files#r514601992?

> ### Code related to MLflow:
> * [`mlflow.pytorch.pytorch_autolog`]
> This is an experimental api that logs ML model artifacts and metrics.
> The metrics are logged during training of the model.

Reviewer: ```suggestion
```

I think we can remove this section given the rewording suggested in https://github.com/mlflow/mlflow/pull/3608/files#r514601992

> ## Setting tracking URI

Reviewer: ```suggestion
## Logging to a custom tracking server

To configure MLflow to log to a custom (non-default) tracking location, set the ``MLFLOW_TRACKING_URI`` environment variable, e.g. via  ``export MLFLOW_TRACKING_URI=http://localhost:5000/``.  For more details, see [the docs](https://mlflow.org/docs/latest/tracking.html#where-runs-are-recorded)
```

Could we reword this section as suggested? And move it to the end of the README? I think it'll only be relevant to a small number of readers
